### ISSUE TITLE
M4 issue: closing the application has extra delay

### ISSUE THREAD

**Issue by ypujante** [JetBrains/compose-jb#748]
As can be seen in this animation:

![9sk77vgi](https://user-images.githubusercontent.com/357116/120683461-7db04b00-c452-11eb-9770-b7f36076b483.gif)

there is about a 2-3 seconds delay after the window is closed and the application terminates.

This is using M4 with the new window api

```
fun main() {

  application {
    val state = rememberWindowState(size = userPreference.windowSize)

    Window(state = state,
      title = "Workout Timer",
      resizable = true,
      onCloseRequest = {
        workoutTimer.reset()
        state.isOpen = false
      }
    ) {
      App(workoutTimer, userPreference)
    }
  }

  println("terminated")
}
```

The message "terminated" is printed instantly when closing the window, but there is a lag to terminate the process. If you select the menu/quit (or command-q on macOS), then the termination is instantaneous.

There was no such delay in M3.

This extra delay is annoying in development because I run the application from the IDE and usually change the code and when I want to restart the app, I close it but then I have to wait for the "run" button to be available again which only happens when the application actually terminates...

**Comment by igordmn** [comment]
It is because in the old API we used `exitProcess(0)` when all windows are closed.

This is not always correct, because in this case we didn't call dispose/close methods of some native resources (all threads will be terminated without calling `finally` block of `try-finally` statement).

If you want a fast exit, you can call `exitProcess(0)` manually after `application { }`, but you should be sure that all run threads of your application can be terminated in a fast-way. Note that coroutines that are run inside `LaunchedEffect` in any Composable will be already finished after `application {}` block, so you don't have to be worry about them.

Not sure yet, but probably now application terminates longer because of the Swing resources.

I will investigate, can we call `exitProcess(0)` implicitly inside `application`, or tweak Swing to exit faster.

**Comment by igordmn** [comment]
Should be fixed in 1.0.0-beta6-dev455.

Now we call `exitProcess` in the end of the `application`

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-6231">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
